api/controllers/standard/standard.controller.js

```diff
@@ -1,15 +1,40 @@
 var Standard = require('../../models/standard.model');
 
-var create = (req, res, next) => {
-    var newStandard = new Standard(req.body);
-
-    newStandard.save(function (err) {
-        if (err)
-            return next(err);
-        res.json({
-            message: 'User created'
-        });
-    });
+var create = async (req, res, next) => {
+    let data = req.body;
+    const shortName = data.info.shortName;
+    await Standard.find({"info.shortName":shortName})
+        .then(async result => {
+            if(result.length>0){
+                res.json({status:409,message:"Duplicate entry found"})
+            }else{
+                {
+                    const api_auth_key = data.api_auth_key;
+                    delete data['api_auth_key'];
+                    if(process.env.api_auth_key === api_auth_key){
+                        await Standard.create(data)
+                            .then(result => {
+                                res.json({
+                                    status:200,
+                                    message: 'Insert Sucessfull'
+                                });
+                            })
+                            .catch(err => {
+                                res.json({
+                                    status:404,
+                                    message:"Insert Unsucesessfull",
```

Comment:
A Better message could be: 'Standard creation failed'

Reply:
Noted